### DISCUSSION TITLE
Request unit test guidance

### DISCUSSION THREAD

**Discussion by egieske** [riscv-software-src/riscv-perf-model#98] [Q&A] [answered]
We have a basic L2Cache unit for addition to Olympia that correctly runs with our traces. We intend to develop a unit test for it. Also, Olympia style guidelines require a regression test of the added feature(s). Is there an good example of a unit test and/or a regression test to guide our test development? Regardless of any example(s), please give guidance on test development for new Olympia units.

**Comment by arupc** [comment]
An example of unit test is available in PR#92 (unit tests for LSU).

@klingaard any thing else you would recommend?

**Reply by arupc** [reply, marked as answer]
This may be helpful as well: https://github.com/riscv-software-src/riscv-perf-model/issues/4#issuecomment-1546693400
